app/src/main/java/de/westnordost/streetcomplete/util/math/FlatEarthMath.kt

```diff
@@ -45,15 +52,21 @@ fun LatLon.flatDistanceToArcs(polyLine: List<LatLon>, globeRadius: Double = EART
         .minOf { flatDistanceToArc(it.first, it.second, globeRadius) }
 }
 
+/** Returns the approximate distance of two points on a sphere.
+ *  Takes into account wrapping at 180th longitude */
 private fun flatAngularDistance(φ1: Double, λ1: Double, φ2: Double, λ2: Double): Double {
     // https://en.wikipedia.org/wiki/Geographical_distance#Spherical_Earth_projected_to_a_plane
     val δφ = φ1 - φ2
-    val δλ = λ1 - λ2
-    val cosδλ = approxCos((φ1 + φ2) / 2) * δλ
+    var δλ = abs(λ1 - λ2)
+    if (δλ > PI)
+        δλ = 2 * PI - δλ
+    val cosδλ = approximateCos((φ1 + φ2) / 2) * δλ
```

Comment:
This won't work if λ1 is e.g. than 5 * PI or -5 * PI. The longitude of a `LatLon` is not necessarily normalized. If you want to guarantee the value to be in a certain co-domain (e.g. -PI .... PI), have a look at `normalizeLongitude` (but that one works with degrees, not radians).

Also, the `abs(...)` looks suspect, because you truncate half of the co-domain of the result. Or was this a bug before that you corrected? (Better make sure all is covered in tests.)

Reply:
>  The longitude of a LatLon is not necessarily normalized

I thought it must be:
https://github.com/streetcomplete/StreetComplete/blob/14b80d7c5e1335d859c219e1735a238f854215e2/app/src/main/java/de/westnordost/streetcomplete/data/osm/mapdata/Element.kt#L85-L89

> Also, the abs(...) looks suspect [...]

Since only the square of `δλ` is actually used, we don't need to care about the sign.
I added the `abs` instead of `else if (δλ < -PI) δλ = -2 * PI + δλ`

> Better make sure all is covered in tests

What kind of further tests do you have in mind? Other than delivering very similar results to the spherical functions if distance is small...

Reply:
Oh, you are right. 

> What kind of further tests do you have in mind?

I meant if the `abs` thing was a bugfix. But I just didn't understand it, so that's fine.